src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -196,68 +231,74 @@ impl StaticHeader {
     }
 
     /// Try to find a valid StaticHeader on a device.
-    /// If there is no StaticHeader on the device, return None.
-    /// If there is a problem reading a header, return an error.
+    /// Return None if the static header's magic does not match for *both* copies.
     fn setup<F>(f: &mut F) -> StratisResult<Option<StaticHeader>>
     where
-        F: Read + Seek,
+        F: Read + Seek + SyncAll,
     {
-        #![allow(unused_io_amount)]
-        f.seek(SeekFrom::Start(0))?;
-        let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
-
-        // FIXME: See https://github.com/stratis-storage/stratisd/pull/476
-        f.read(&mut buf)?;
-
-        // TODO: repair static header if one incorrect?
-        // Note: this would require some adjustment or some revision to
-        // setup_from_buf().
-        StaticHeader::setup_from_buf(&buf)
-    }
-
-    /// Try to find a valid StaticHeader in a buffer.
-    /// If there is an error in reading the first, try the next. If there is
-    /// no error in reading the first, assume it is correct, i.e., do not
-    /// verify that it matches the next.
-    /// Return None if the static header's magic number is wrong.
-    fn setup_from_buf(buf: &[u8; _BDA_STATIC_HDR_SIZE]) -> StratisResult<Option<StaticHeader>> {
-        let sigblock_spots = [
-            &buf[SECTOR_SIZE..2 * SECTOR_SIZE],
-            &buf[9 * SECTOR_SIZE..10 * SECTOR_SIZE],
-        ];
-
-        for buf in &sigblock_spots {
-            match StaticHeader::sigblock_from_buf(buf) {
-                Ok(val) => return Ok(val),
-                _ => continue,
+        let buf = BDA::read(f)?;
+        let buf_loc_1 = &buf[SECTOR_SIZE..2 * SECTOR_SIZE];
+        let buf_loc_2 = &buf[9 * SECTOR_SIZE..10 * SECTOR_SIZE];
+
+        match (
+            StaticHeader::sigblock_from_buf(buf_loc_1),
+            StaticHeader::sigblock_from_buf(buf_loc_2),
+        ) {
+            (Ok(loc_1), Ok(loc_2)) => {
+                match (loc_1, loc_2) {
+                    (Some(loc_1), Some(_)) => {
```

Comment:
Since both values are read it costs virtually nothing to verify that they are equal. So it makes sense to do that and return an error if they are not (because you do not know which one is correct, and one must be false). You'll have to derive ```Eq```, ```PartialEq``` for ```StaticHeader```, but I don't see anything wrong with that.

Reply:
They both have a time stamp, should we return the newer of the two?

Reply:
Returning most recent isn't unreasonable.